docs/csharp/misc/cs1929.md

```diff
@@ -10,44 +10,98 @@ ms.assetid: effdd5d4-e156-418b-9d45-4ca194ab4319
 ---
 # Compiler Error CS1929
 
-Instance argument: cannot convert from 'typeA' to 'typeB'.  
-  
- This error is generated when you try to invoke an extension method from a class that it does not extend. In the example shown here, the extension method is defined for the derived class `A`, but not for the base class `B`.  
-  
-## To correct this error  
-  
-1. Create a new extension method for the type where you have to invoke it, or else move the call into an object of the type that the existing method extends.  
-  
-## Example  
-
- The following code generates CS1928 and CS1929:  
-  
-```csharp  
-// cs1929.cs  
-using System.Linq;  
-    using System.Collections;  
-  
-    static class Ext  
-    {  
-        public static void ExtMethod(this A a)  
-        {  
-        }  
-    }  
-  
-    class A : B  
-    {  
-    }  
-  
-    class B  
-    {  
-        static void Main()  
-        {  
-            B b = new B();  
-            b.ExtMethod(); // CS1929  
-        }  
-    }  
-```  
-  
+'typeB' does not contain a definition for 'method' and the best extension method overload 'typeC.method' requires a receiver of type 'typeA'
+
+This error is generated when you try to invoke an extension method from a class that it does not extend. In the example shown here, the extension method is defined for the derived class `A`, but not for the base class `B`.  
```

Comment:
I'll make suggestions throughout this sample. I'd prefer following the convention of `B` for the base class and `D` for the derived class:

```suggestion
This error is generated when you try to invoke an extension method from a class that it does not extend. In the example shown here, the extension method is defined for the derived class `D`, but not for the base class `B`.  
```